core/src/main/scala/io/iteratee/Enumeratee.scala

```diff
@@ -82,6 +82,34 @@ final object Enumeratee {
       }
     }
 
+  final def tailRecM[F[_], A, B](f: A => Enumerator[F, Either[A, B]])(implicit F: Monad[F]): Enumeratee[F, Either[A, B], B] =
+    new PureLoop[F, Either[A, B], B] {
+      protected final def loop[C](step: Step[F, B, C]): Step[F, Either[A, B], Step[F, B, C]] =
+        new StepCont[F, Either[A, B], B, C](step) {
+          @annotation.tailrec
+          final def feedEl(e: Either[A, B]): F[Step[F, Either[A, B], Step[F, B, C]]] = {
+            e match {
+              case Left(a) =>
+                f(a) match {
+                  case Enumerator.EnumOne(e, _) => feedEl(e)
+                  case other => other(this)
```

Comment:
this could potentially blow up. Ideally we don't pass `this` around. If `Enumerator` were sealed, we could probably do that. Otherwise, we could make case classes for some common enumerators so we can pattern match here and manually call the loop here.